src/libstore/misc.cc

```diff
@@ -351,4 +360,112 @@ OutputPathMap resolveDerivedPath(Store & store, const DerivedPath::Built & bfd,
     return outputs;
 }
 
+
+SingleDerivedPath tryResolveDerivedPath(Store & store, const SingleDerivedPath & req, Store * evalStore_)
+{
+    auto & evalStore = evalStore_ ? *evalStore_ : store;
+
+    return std::visit(overloaded {
+        [&](const SingleDerivedPath::Opaque &) -> SingleDerivedPath {
```

Comment:
@roberth suggests: We may want to rename this to `DerivablePath` as it's not derived yet, and also the constructors to `Constant` and `Output`.